### ISSUE TITLE
[feature request] Add training={True|False} kwarg to nn.RNN.forward(), nn.Sequential.forward()

### ISSUE THREAD

**Issue by Fenugreek** [pytorch/pytorch#5908]
Both the RNN Module subclass and Sequential Module subclass support dropout layers during construction, but it is currently bit of a gotcha to switch the dropout off when in evaluation mode.

Can we have a simpler and more visible way to switch off dropout when not training, by for example having a `training=True` (default) kwarg to the `.forward()` method, similar to the kwarg in `nn.functional.dropout()`?

Currently, for the RNN subclasses one has to do the following:

```
class myNN(nn.Module):

    def __init__(self, n_layers=2, dropout=0.3):

        super(myNN, self).__init__()
	self.gru = nn.GRU(100, 500, n_layers, dropout=dropout)
        ...

    def forward(self, inputs, hidden, training=True):

	self.gru.training = training
        output = self.gru(inputs, hidden)
        ...
```

And for the Sequential subclass, I think one has to explicitly reference all the layer(s) in the module list and set their `.training` attributes to `False`.

**Comment by apaszke** [comment]
Adding a `training` kwarg would be a very strong requirement, and I believe this is unnecessary. You can just do `model.train()` and `model.eval()` to switch *all* modules (including submodules) to train/eval mode. Alternatively, if you want to do this selectively, you can index out the relevant submodules and do that only to them. Hope this helps.
